src/main/java/com/redislabs/redisai/Model.java

```diff
@@ -0,0 +1,152 @@
+package com.redislabs.redisai;
+
+import com.redislabs.redisai.exceptions.JRedisAIRunTimeException;
+import redis.clients.jedis.util.SafeEncoder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Direct mapping to RedisAI Model
+ */
+public class Model {
+    private Backend backend;
+    private Device device;
+    private String[] inputs;
+    private String[] outputs;
+    private byte[] blob;
+    private String tag;
+
+    /**
+     * @param backend - the backend for the model. can be one of TF, TFLITE, TORCH or ONNX
+     * @param device  - the device that will execute the model. can be of CPU or GPU
+     * @param inputs  - one or more names of the model's input nodes (applicable only for TensorFlow models)
+     * @param outputs - one or more names of the model's output nodes (applicable only for TensorFlow models)
+     * @param blob    - the Protobuf-serialized model
+     */
+    public Model(Backend backend, Device device, String[] inputs, String[] outputs, byte[] blob) {
+        this.backend = backend;
+        this.device = device;
+        this.inputs = inputs;
+        this.outputs = outputs;
+        this.blob = blob;
+        this.tag = "";
```

Comment:
@filipecosta90 are you sure the default should be an empty string and not null?

Reply:
solved on the new commit